triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -614,14 +629,55 @@ private[lf] class Runner private (
     var numberOfExerciseByKey = 0
 
     val startTime = System.nanoTime()
+    val ruleEvaluationTimer = if (triggerConfig.hardLimit.allowTriggerTimeouts) {
+      materializer.scheduleOnce(
+        triggerConfig.hardLimit.ruleEvaluationTimeout,
+        () => {
+          triggerContext.logError(
+            "Stopping trigger as the rule evaluator has exceeded its allotted running time",
+            "rule-evaluation-timeout" -> triggerConfig.hardLimit.ruleEvaluationTimeout,
+          )
+
+          hardLimitKillSwitch.abort(
+            TriggerRuleEvaluationTimeout(triggerConfig.hardLimit.ruleEvaluationTimeout)
+          )
```

Comment:
Need to use a kill switch as the scheduler  thread needs to communicate with the streaming async workflow